### PR TITLE
Move TV Library to its own view

### PR DESCRIPTION
**Pull request by 1hitsong**
**Changes**
Moves TV Libraries off the shared itemgrid view and into its own TVLibraryView component.

This allows us to avoid needing to have collection type checks all over the view and allows us to modify the tv library view without fear of messing up anything else.

As a followup, cleanup item, we should remove any TV library conditional code from itemgrid

### CODE REVIEW

**Comment by cewert** [comment]
I know this doesn't say merge conflict but the required workflow won't run until you merge the latest version of unstable into your PR branch FYI

**Comment by jimdogx** [comment]
There is one warning, but otherwise appears to work great.

```
[{
	"resource": "/Users/jimi/Source/jellyfin-1hitsong/components/ItemGrid/TVShowLibraryView.xml",
	"owner": "_generated_diagnostic_collection_name_#0",
	"code": "1107",
	"severity": 4,
	"message": "This import is unnecessary because compiler option 'autoImportComponentScript' is enabled",
	"source": "brs",
	"startLineNumber": 37,
	"startColumn": 41,
	"endLineNumber": 37,
	"endColumn": 62
}]
```

File: components/ItemGrid/TVShowLibraryView.brs [hunk 1]
```diff
@@ -0,0 +1,740 @@
...
+    m.loadItemsTask.totalRecordCount = 0
+
+    m.spinner = m.top.findNode("spinner")
+    m.spinner.visible = true
+
+    m.Alpha = m.top.findNode("AlphaMenu")
+    m.AlphaSelected = m.top.findNode("AlphaSelected")
+
+    'Get reset folder setting
+    m.resetGrid = get_user_setting("itemgrid.reset") = "true"
```
Comment (cewert) [inline components/ItemGrid/TVShowLibraryView.brs:68]: using old syntax

File: components/ItemGrid/TVShowLibraryView.brs [hunk 2]
```diff
@@ -0,0 +1,740 @@
...
+
+    'Check if device has voice remote
+    devinfo = CreateObject("roDeviceInfo")
+    m.deviFeature = devinfo.HasFeature("voice_remote")
```
Comment (cewert) [inline components/ItemGrid/TVShowLibraryView.brs:72]: this is saved to global device - `m.global.device`

File: components/ItemGrid/TVShowLibraryView.brs [hunk 3]
```diff
@@ -0,0 +1,740 @@
...
+    end if
+
+    if m.top.parentItem.backdropUrl <> invalid
+        SetBackground(m.top.parentItem.backdropUrl)
+    end if
+
+    ' Read view/sort/filter settings
+    if m.top.parentItem.collectionType = "livetv"
+        ' Translate between app and server nomenclature
+        viewSetting = get_user_setting("display.livetv.landing")
```
Comment (cewert) [inline components/ItemGrid/TVShowLibraryView.brs:105]: old syntax

File: components/ItemGrid/TVShowLibraryView.brs [hunk 4]
```diff
@@ -0,0 +1,740 @@
...
+            m.view = "livetv"
+        end if
+        m.sortField = get_user_setting("display.livetv.sortField")
+        sortAscendingStr = get_user_setting("display.livetv.sortAscending")
+        m.filter = get_user_setting("display.livetv.filter")
+    else
+        m.sortField = get_user_setting("display." + m.top.parentItem.Id + ".sortField")
+        sortAscendingStr = get_user_setting("display." + m.top.parentItem.Id + ".sortAscending")
+        m.filter = get_user_setting("display." + m.top.parentItem.Id + ".filter")
+        m.view = get_user_setting("display." + m.top.parentItem.Id + ".landing")
```
Comment (cewert) [inline components/ItemGrid/TVShowLibraryView.brs:126]: old syntax

File: components/ItemGrid/TVShowLibraryView.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    <field id="imageDisplayMode" type="string" value="scaleToZoom" />
+    <field id="AlphaSelected" type="string" alias="AlphaMenu.itemAlphaSelected" alwaysNotify="true" onChange="onItemAlphaSelected" />
+    <field id="alphaActive" type="boolean" value="false" />
+    <field id="jumpToItem" type="integer" value="" />
+  </interface>
+  <script type="text/brightscript" uri="pkg:/source/utils/misc.brs" />
+  <script type="text/brightscript" uri="pkg:/source/utils/config.brs" />
+  <script type="text/brightscript" uri="pkg:/source/api/baserequest.brs" />
+  <script type="text/brightscript" uri="pkg:/source/utils/deviceCapabilities.brs" />
+  <script type="text/brightscript" uri="TVShowLibraryView.brs" />
```
Comment (cewert) [inline components/ItemGrid/TVShowLibraryView.xml:37]:
```suggestion
```

This is the line from the warning. It's not needed.

File: components/ItemGrid/TVShowLibraryView.xml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    <field id="quickPlayNode" type="node" alwaysNotify="true" />
+    <field id="imageDisplayMode" type="string" value="scaleToZoom" />
+    <field id="AlphaSelected" type="string" alias="AlphaMenu.itemAlphaSelected" alwaysNotify="true" onChange="onItemAlphaSelected" />
+    <field id="alphaActive" type="boolean" value="false" />
+    <field id="jumpToItem" type="integer" value="" />
+  </interface>
+  <script type="text/brightscript" uri="pkg:/source/utils/misc.brs" />
+  <script type="text/brightscript" uri="pkg:/source/utils/config.brs" />
+  <script type="text/brightscript" uri="pkg:/source/api/baserequest.brs" />
+  <script type="text/brightscript" uri="pkg:/source/utils/deviceCapabilities.brs" />
```
Comment (cewert) [inline components/ItemGrid/TVShowLibraryView.xml:36]: These should be import statements in `TVShowLibraryView.brs`

**Review by cewert — CHANGES_REQUESTED** [review]

**Pull request closed without merging**